src/index.tsx

```diff
@@ -14,7 +14,7 @@ export interface IReactToPrintProps {
     /** Copy styles over into print window. default: true */
     copyStyles?: boolean;
     /** Callback function to trigger before print */
-    onBeforePrint?: () => void;
+    onBeforePrint?: () => void | Promise<any>;
```

Comment:
Please also update the README

Reply:
@MatthewHerbst Done. Please tell me if the explanation is good enough.